src/hotspot/share/code/aotCodeCache.cpp

```diff
@@ -926,18 +926,13 @@ AOTCodeEntry* AOTCodeCache::find_code_entry(const methodHandle& method, uint com
   if (is_on() && _cache->cache_buffer() != nullptr) {
     ResourceMark rm;
     const char* target_name = method->name_and_sig_as_C_string();
-    uint hash = java_lang_String::hash_code((const jbyte*)target_name, (int)strlen(target_name));
+    uint hash = (uint)pointer_delta((address)method(), (address)SharedBaseAddress, 1);
```

Comment:
Can you add AOTCacheAccess method for calculating delta during product run? It should be together with other AOTCacheAccess  methods we use here.

Reply:
ok, I will add an api in AOTCacheAccess.